include/engine/datafacade/shared_memory_datafacade.hpp

```diff
@@ -40,58 +38,49 @@ class SharedMemoryDataFacade : public ContiguousInternalMemoryDataFacadeBase
+            data_region == storage::REGION_1 ? shared_barriers->region_1_mutex
+                                             : shared_barriers->region_2_mutex,
             boost::interprocess::defer_lock);
 
         // if this returns false this is still in use
         if (current_regions_lock.try_lock() && exclusive_lock.try_lock())
         {
             if (storage::SharedMemory::RegionExists(data_region))
             {
-                BOOST_ASSERT(storage::SharedMemory::RegionExists(layout_region));
-
-                auto shared_regions = storage::makeSharedMemory(storage::CURRENT_REGIONS);
+                auto shared_region = storage::makeSharedMemory(storage::CURRENT_REGION);
                 const auto current_timestamp =
-                    static_cast<const storage::SharedDataTimestamp *>(shared_regions->Ptr());
+                    static_cast<const storage::SharedDataTimestamp *>(shared_region->Ptr());
 
                 // check if the memory region referenced by this facade needs cleanup
-                if (current_timestamp->data == data_region)
+                if (current_timestamp->region == data_region)
                 {
-                    BOOST_ASSERT(current_timestamp->layout == layout_region);
                     util::Log(logDEBUG) << "Retaining data with shared timestamp "
                                         << shared_timestamp;
                 }
                 else
                 {
                     storage::SharedMemory::Remove(data_region);
-                    storage::SharedMemory::Remove(layout_region);
                 }
             }
         }
     }
 
     SharedMemoryDataFacade(const std::shared_ptr<storage::SharedBarriers> &shared_barriers_,
-                           storage::SharedDataType layout_region_,
                            storage::SharedDataType data_region_,
                            unsigned shared_timestamp_)
-        : shared_barriers(shared_barriers_), layout_region(layout_region_),
-          data_region(data_region_), shared_timestamp(shared_timestamp_)
+        : shared_barriers(shared_barriers_), data_region(data_region_), shared_timestamp(shared_timestamp_)
     {
         util::Log(logDEBUG) << "Loading new data with shared timestamp " << shared_timestamp;
 
-        BOOST_ASSERT(storage::SharedMemory::RegionExists(layout_region));
-        m_layout_memory = storage::makeSharedMemory(layout_region);
-
         BOOST_ASSERT(storage::SharedMemory::RegionExists(data_region));
         m_large_memory = storage::makeSharedMemory(data_region);
 
-        InitializeInternalPointers(*reinterpret_cast<storage::DataLayout *>(m_layout_memory->Ptr()),
-                                   reinterpret_cast<char *>(m_large_memory->Ptr()));
+        InitializeInternalPointers(*reinterpret_cast<storage::DataLayout *>(m_large_memory->Ptr()),
+                                   reinterpret_cast<char *>(m_large_memory->Ptr()) + sizeof(storage::DataLayout));
```

Comment:
Padding depends on packing, but it does not matter here.

Alignment will be adjusted at https://github.com/Project-OSRM/osrm-backend/blob/master/include/storage/shared_datatype.hpp#L145 internally in the data block.
It will work because GetSizeOfLayout pessimistically assumes misaligned data and request ~500 bytes overhead to adjust pointers